File: DS lab external/binarySearch.c
#include<stdio.h>
int binarySearch(int arr[] , int size  , int ele)
{
    int left = 0;
    int right = size-1;

    while(left <= right)
    {

        int mid = (left+right)/2;
        // printf("mid ==  %d\n ",mid);
        if(ele < arr[mid])
            right = mid-1;
        else if(ele > arr[mid])
            left = mid+1;
        else if(ele == arr[mid])
            return mid+1;
    }

    return 0;
}

int main()
{

    int arr[10] = {1,2,3,4,5,6,7,8,9,0};

    int size = 10;
    int ele = 11;
    int res = binarySearch(arr , size , ele);

    if(res)
        printf("Found at %d position \n" , res);
    else printf("Not found\n");
    printf("%d\n" , res);

}

File: searching Algos/binarySearch.py
index = None
def binarySort(arr,key):
    low = 0                 #0
    high = len(arr)-1       #8
    
    while low <= high:       #5 < 5
        mid = (low+high)//2 #mid = 6
        if key < arr[mid]:  #67 < 78
            high = mid-1    #high = 5
        elif key > arr[mid]:#67 > 56
            low = mid+1     #   low = 5
        elif key == arr[mid]: #
            return mid
            
key = 90
arr= [12,23,34,45,56,67,78,89,90]
index = binarySort(arr,key)
if index==None:
    print("Not found")
else:
    print( key,"found at index  : ",index)
